### PR TITLE
Sellers can edit products

### PR DESCRIPTION
**Pull request by ethanschreur**
**What does this PR do?**
- This pull request allows sellers to edit product data

**Description of the task to be completed** 
- Click on a products row to go to a page with that product's information
- Show how the product information changes the product data's visual representation
- When sellers edit product information, update the visual representation

**How should this  be tested**
- (On Windows) Run `. venv/scripts/activate` 
- `python run.py`
- Go to http://127.0.0.1:5000/login
- Enter the correct email and password and submit them
- View the dashboard page
- Press the products button
- Fill out the "add products" form and submit it
- Click on the added product in the products table
- Edit the product data in the "Edit Products" section
- See the changes in the visual representation
- Go back to the products page and view the changes there.
- Click on the same product and add a subproduct
- View that subproduct data in the Edit Subproducts section
- Confirm that the subproduct image is added to the visual representation
- Edit that subproduct's data and confirm that changes were made to the visual representation

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -44,19 +44,61 @@ def products():
...
+    if ("seller_email" not in session):
+        return redirect('/login')
+    new_subproduct = Subproduct(
+        product_id = id,
+        name = request.form['subproduct_name'],
+        image_url = request.form['subproduct_image']
+    )
+    db.session.add(new_subproduct)
+    db.session.commit()
+    return redirect(f'/products/{id}')
```
Comment (alexkayabula) [inline app/__init__.py:93]: @ethanschreur I think this should be in a different PR. I don't see how it is related to editing products.What do you think ?

File: app/__init__.py [hunk 2]
```diff
@@ -44,19 +44,61 @@ def products():
...
-    return redirect('/products') 
\ No newline at end of file
+    return redirect('/products') 
+
+@app.route('/products/<id>', methods=['GET'])
+def get_product(id):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    product = Product.query.get_or_404(id)
+    return render_template('seller/product.html', prod = product)
```
Comment (alexkayabula) [inline app/__init__.py:67]: @ethanschreur How is this related to editing products ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: app/__init__.py [hunk 3]
```diff
@@ -44,19 +44,61 @@ def products():
...
+@app.route('/products/<id>/subproducts/<sid>', methods=['POST'])
+def update_subproducts(id, sid):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    subproduct = Subproduct.query.get(sid)
+    subproduct.name = request.form['subproduct_name']
+    subproduct.image_url = request.form['subproduct_image']
+    db.session.add(subproduct)
+    db.session.commit()
+    return redirect(f'/products/{id}')
```
Comment (alexkayabula) [inline app/__init__.py:104]: @ethanschreur This PR is meant to allow users to update products. I think this should be in a PR of its owm

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: app/__init__.py (hunk 1 above)
Comment (ethanschreur) [reply app/__init__.py:93]: In my mind, part of editing a product includes adding subproducts to the product.

File: app/__init__.py (hunk 2 above)
Comment (ethanschreur) [reply app/__init__.py:67]: That is the route that leads to where the products can be edited.

File: app/__init__.py (hunk 3 above)
Comment (ethanschreur) [reply app/__init__.py:104]: In my mind, part of editing a product is editing additional information about the product (like what subproducts are part of the product).

File: app/__init__.py (hunk 1 above)
Comment (ethanschreur) [reply app/__init__.py:93]: > @ethanschreur I think this should be in a different PR. I don't see how it is related to editing products.What do you think ?

In my mind, part of editing a product includes adding subproducts to the product.

File: app/__init__.py (hunk 2 above)
Comment (ethanschreur) [reply app/__init__.py:67]: > @ethanschreur How is this related to editing products ?

That is the route that leads to where the products can be edited.

File: app/__init__.py (hunk 3 above)
Comment (ethanschreur) [reply app/__init__.py:104]: > @ethanschreur This PR is meant to allow users to update products. I think this should be in a PR of its owm

In my mind, part of editing a product is editing additional information about the product (like what subproducts are part of the product).

File: app/__init__.py (hunk 1 above)
Comment (alexkayabula) [reply app/__init__.py:93]: @ethanschreur being able to add a subproduct to a product its a feature on its own. Editing/Updating an already added subproduct is also a feature on its own.

File: app/__init__.py (hunk 3 above)
Comment (alexkayabula) [reply app/__init__.py:104]: @ethanschreur being able to add a subproduct to a product its a feature on its own. Editing/Updating an already added subproduct is also a feature on its own.

File: app/__init__.py (hunk 2 above)
Comment (alexkayabula) [reply app/__init__.py:67]: That's ok

File: app/__init__.py (hunk 1 above)
Comment (ethanschreur) [reply app/__init__.py:93]: Gotcha. I will make the changes.

Comment (ethanschreur) [reply app/__init__.py:93]: > @ethanschreur being able to add a subproduct to a product its a feature on its own. Being able to Edit/Update a already added subproduct is also a feature on its own.

I will make the changes

File: app/__init__.py (hunk 3 above)
Comment (ethanschreur) [reply app/__init__.py:104]: > @ethanschreur being able to add a subproduct to a product its a feature on its own. Being able to Edit/Update a already added subproduct is also a feature on its own.

I will make the changes

**Comment by ethanschreur** [comment]
Closed in order to separate this PR out into additional PRs.
